src/main/java/com/metamx/collections/bitmap/RoaringBitmapFactory.java

```diff
@@ -58,9 +61,7 @@ public MutableBitmap makeEmptyMutableBitmap()
   @Override
   public ImmutableBitmap makeEmptyImmutableBitmap()
   {
-    return new WrappedImmutableRoaringBitmap(
-        EMPTY_IMMUTABLE_BITMAP
-    );
+    return WRAPPED_IMMUTABLE_ROARING_BITMAP;
```

Comment:
@drcrallen 

Yes, there is just one "PointableRoaringArray highLowContainer" per ImmutableRoaringBitmap instance. It is an array of containers (hence the name "array") where the data is stored. So this is as should be. The getIntIterator runs on top of container iterator that is generated from the call to "getContainerPointer()" which, itself, returns a new object as you can see there :

https://github.com/lemire/RoaringBitmap/blob/master/src/main/java/org/roaringbitmap/buffer/ImmutableRoaringArray.java#L215

We use so many layers of abstraction because we get MutableRoaringBitmap and ImmutableRoaringBitmap to share much of the same code, while having different physical implementations.

I would certainly consider the fact that you can't iterate in a thread-safe manner over an immutable bitmap as a critical bug. It is not something we would tolerate knowingly in the code.